hawkular-inventory-api/src/main/java/org/hawkular/inventory/api/model/InventoryStructure.java

```diff
@@ -0,0 +1,847 @@
+     */
+    enum EntityType {
+        //the order is significant.. the latter cannot exist without (some of) the prior
+        feed(Feed.class, Feed.Blueprint.class, SegmentType.f),
+        resourceType(ResourceType.class, ResourceType.Blueprint.class, SegmentType.rt),
+        metricType(MetricType.class, MetricType.Blueprint.class, SegmentType.mt),
+        operationType(OperationType.class, OperationType.Blueprint.class, SegmentType.ot),
+        metric(Metric.class, Metric.Blueprint.class, SegmentType.m),
+        resource(Resource.class, Resource.Blueprint.class, SegmentType.r),
+        dataEntity(DataEntity.class, DataEntity.Blueprint.class, SegmentType.d);
+
+        public final Class<? extends Entity<?, ?>> elementType;
+        public final Class<? extends Entity.Blueprint> blueprintType;
+        public final SegmentType segmentType;
+
+        public static EntityType of(Class<?> type) {
+            for (EntityType t : EntityType.values()) {
+                if (type.equals(t.elementType)) {
+                    return t;
+                }
+            }
+
+            throw new IllegalArgumentException("Unsupported type of entity: " + type);
+        }
+
+        public static EntityType of(SegmentType seg) {
+            for (EntityType t : EntityType.values()) {
+                if (seg == t.segmentType) {
+                    return t;
+                }
+            }
+
+            throw new IllegalArgumentException("Unsupported type of path segment: " + seg);
+        }
+
+        public static boolean supports(SegmentType seg) {
+            for (EntityType t : EntityType.values()) {
+                if (seg == t.segmentType) {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        public static EntityType ofBlueprint(Class<?> type) {
+            for (EntityType t : EntityType.values()) {
+                if (type.equals(t.blueprintType)) {
+                    return t;
+                }
+            }
+
+            return null;
+        }
+
+        EntityType(Class<? extends Entity<?, ?>> elementType, Class<? extends Entity.Blueprint> blueprintType,
+                   SegmentType segmentType) {
+            this.elementType = elementType;
+            this.blueprintType = blueprintType;
+            this.segmentType = segmentType;
+        }
+    }
+
+    /**
+     * Represents the structure of the inventory off-line, without access to an inventory instance. This implies that
+     * an instance holds on to all data required and thus can occupy a lot of memory. On the other hand, it is
+     * serializable.
+     * <p>
+     * This is not directly instantiable but rather can be either copied from another structure (possibly lazily
+     * loaded) or built using a {@link Builder}.
+     */
+    class Offline<Root extends Entity.Blueprint> implements InventoryStructure<Root>, Serializable {
```

Comment:
Does it mean, we can do some kind of snapshots of the inventory structure and save/load it.